src/http-api/resources/object.js

```diff
@@ -493,8 +450,12 @@ exports.patchRmLink = {
   handler: (request, reply) => {
     const root = request.pre.args.root
     const link = request.pre.args.link
+    const ipfs = request.server.app.ipfs
 
-    request.server.app.ipfs.object.patch.rmLink(root, link, (err, node) => {
+    waterfall([
+      (cb) => ipfs.object.patch.rmLink(root, link, cb),
+      (node, cb) => node.toJSON(cb)
+    ], (err, res) => {
```

Comment:
Seems overly verbose, and I really dislike all caps naming. Not sure about a better name